client/app/cloud/project/compute/snapshot/cloud-project-compute-snapshot.controller.js

```diff
@@ -60,29 +60,42 @@ angular.module("managerApp")
         };
     }
     self.snapshotPriceStruct = {
-        prices:[],
+        prices:null,
         size:0,
         total:{}
     };
 
 
     function setPrice(){
-        var totalSize = 0,
-        total = {};
-        angular.forEach(self.table.snapshot, function(value){
-            totalSize += value.size;
-            value.price = getPrice(value.size);
+        let totalSize = 0,
+        totalPrice = 0;
+        if (!self.table.snapshot.length) {
+            return;
+        }
+        angular.forEach(self.table.snapshot, function(snapshot){
+            totalSize += snapshot.size;
+            snapshot.price = getMonthlyPrice(snapshot.size, snapshot.planCode);
+            if (snapshot.price) {
+                totalPrice += snapshot.price.value;
+            }
         });
-        total.price = getPrice(totalSize);
         self.snapshotPriceStruct.size = totalSize;
-        self.snapshotPriceStruct.total = total;
+        // Copy from formatted price to keep currencyCode
+        self.snapshotPriceStruct.total = angular.copy(self.table.snapshot[0].price)
+        self.snapshotPriceStruct.total.value = totalPrice;
+        self.snapshotPriceStruct.total.text = self.snapshotPriceStruct.total.text.replace(/\d+(?:[.,]\d+)?/, "" + totalPrice.toFixed(2))
     }
 
-    function getPrice(size){
+    function getMonthlyPrice(size, planCode){
         // get the price for first location comming
-        var priceStruct = angular.copy(self.snapshotPriceStruct.prices[0].monthlyPrice);
+        let price = self.snapshotPriceStruct.prices[planCode];
+        if (!price) {
+            console.warn("fail to get price", self.snapshotPriceStruct);
```

Comment:
this shouldn't happen... It might be useful for test purposes but it shouldn't happen in prod.